### PR TITLE
Add type=PathArg to remaining FILE args

### PR DESCRIPTION
**Pull request by ishanSrt**
Closes https://github.com/coala/coala/issues/5022

### CODE REVIEW

**Comment by ishanSrt** [comment]
`test_from_clang_fixit` is skipping in my pytest
```
Forbidden skipped test - Skipped: dlsym(RTLD_DEFAULT, clang_annotateTokens): symbol not found. Please ensure that your python bindings are compatible with your libclang.so version.
```

**Comment by userzimmermann** [comment]
@ishanSrt Oh! Just one thing left ;)

Please prefix your commit message with `DefaultArgParser:`

The shortlog will get to long then... But you could replace `remaining` with `all` and somehow save one further character... Then it will fit :)

**Comment by userzimmermann** [comment]
@ishanSrt Or like that... :) Perfect! So... only AppVeyor blocking our way

**Comment by userzimmermann** [comment]
@ishanSrt You actually need to add the code changes from https://github.com/coala/coala/pull/5021 to this PRs commit ;) And there is no need to reference https://github.com/coala/coala/pull/5021 in the commit message

**Comment by ishanSrt** [comment]
>Just add info on those additional changes from #5021 to the commit message body

thats what you told me to do 🧐

**Comment by userzimmermann** [comment]
@ishanSrt Indeed 😄 
And now I upgraded my request ;P

**Comment by userzimmermann** [comment]
@ishanSrt Ok... And now the last task refinement ;) There is no need to reference #5021 in the commit message, but there is still need to explain that additionally those `re.escape`s were removed in the related tests... what I actually meant with _"add info on those additional changes from #5021 to the commit message body"_

**Comment by ishanSrt** [comment]
ok. for a minute I thought that referencing a PR on the commit body accounts for the changes made over there.

**Comment by ishanSrt** [comment]
@userzimmermann how does it look now?

File: tests/settings/SettingTest.py [hunk 1]
```diff
@@ -34,7 +35,7 @@ def test_path(self):
         self.assertEqual(path(self.uut),
                          os.path.abspath(os.path.join('.', '22')))
 
-        abspath = os.path.abspath('.')
+        abspath = PathArg(os.path.abspath('.'))
```
Comment (userzimmermann) [inline tests/settings/SettingTest.py]: Better move this change to #5019 since this test is not related to a cmdline arg

File: tests/settings/ConfigurationGatheringTest.py [hunk 2]
```diff
@@ -70,15 +70,16 @@ def test_gather_configuration(self):
...
+            PathArg(repr(temporary)) +
             ", files : '*.java', save : 'True', test : '5'}")
 
         with make_temp() as temporary:
             sections, local_bears, global_bears, targets = (
                 gather_configuration(*args,
                                      arg_list=['-S test=5',
                                                '-f *.java',
-                                               '-c ' + escape(temporary, '\\'),
+                                               '-c ' + PathArg(temporary),
```
Comment (userzimmermann) [inline tests/settings/ConfigurationGatheringTest.py]: Please just remove `escape` here. No `PathArg`. The latter gets auromatically applied by `DefaultArgParser`

File: tests/settings/ConfigurationGatheringTest.py [hunk 3]
```diff
@@ -70,15 +70,16 @@ def test_gather_configuration(self):
```
Comment (userzimmermann) [inline tests/settings/ConfigurationGatheringTest.py]: the `escape` above also has to be removed. And just like with the other `escape` removal below: No `PathArg`. Just remove

File: tests/settings/ConfigurationGatheringTest.py [hunk 4]
```diff
@@ -70,15 +70,16 @@ def test_gather_configuration(self):
 
         self.assertEqual(
             str(sections['cli']),
-            "cli {bears : 'JavaTestBear', config : " + repr(temporary) +
+            "cli {bears : 'JavaTestBear', config : " +
+            PathArg(repr(temporary)) +
```
Comment (userzimmermann) [inline tests/settings/ConfigurationGatheringTest.py]: Must be `repr(PathArg(temporary))`

This is only coincidentally working because of the remaining `escape(temporary, ...)` above

**Comment by userzimmermann** [comment]
ack https://github.com/coala/coala/pull/5023/commits/aa4a2378c82ac8cbce297348ba2957acb42f878b

**Comment by userzimmermann** [comment]
@ishanSrt Finally 🎉 Very well done :) This PR can be merged now!

**Comment by userzimmermann** [comment]
@ishanSrt Oops! Sorry :/ I accidentally clicked on `[Update branch]` Can you please please rebase and repush your local branch :)

**Comment by userzimmermann** [comment]
reack https://github.com/coala/coala/pull/5023/commits/d975278687bf1de618dec46e5a8865ed8c11a854

**Comment by Makman2** [comment]
@gitmate-bot rebase

**Comment by Makman2** [comment]
Yeah recent build breakage: https://github.com/coala/coala/issues/5045
But fixed now :+1:

**Comment by Makman2** [comment]
@gitmate-bot ff
